go-controller/pkg/node/gateway_iptables.go

```diff
@@ -352,6 +352,96 @@ func getExternalIPTRules(svcPort kapi.ServicePort, externalIP, dstIP string, svc
 	}
 }
 
+func getGatewayForwardRules(svcCIDR *net.IPNet) []iptRule {
+	protocol := getIPTablesProtocol(svcCIDR.IP.String())
+	masqueradeIP := types.V4OVNMasqueradeIP
+	if protocol == iptables.ProtocolIPv6 {
+		masqueradeIP = types.V6OVNMasqueradeIP
+	}
+	return []iptRule{
+		{
+			table: "filter",
+			chain: "FORWARD",
+			args: []string{
+				"-s", svcCIDR.String(),
+				"-j", "ACCEPT",
+			},
+			protocol: protocol,
+		},
+		{
+			table: "filter",
+			chain: "FORWARD",
+			args: []string{
+				"-d", svcCIDR.String(),
+				"-j", "ACCEPT",
+			},
+			protocol: protocol,
+		},
+		{
+			table: "filter",
+			chain: "FORWARD",
+			args: []string{
+				"-s", masqueradeIP,
+				"-j", "ACCEPT",
+			},
+			protocol: protocol,
+		},
+		{
+			table: "filter",
+			chain: "FORWARD",
+			args: []string{
+				"-d", masqueradeIP,
+				"-j", "ACCEPT",
+			},
+			protocol: protocol,
+		},
+	}
+}
+
+func getGatewayDropRules(ifName string) []iptRule {
+	var dropRules []iptRule
+	for _, protocol := range clusterIPTablesProtocols() {
+		dropRules = append(dropRules, []iptRule{
+			{
+				table: "filter",
+				chain: "FORWARD",
+				args: []string{
+					"-i", ifName,
+					"-j", "DROP",
+				},
+				protocol: protocol,
+			},
+			{
+				table: "filter",
+				chain: "FORWARD",
+				args: []string{
+					"-o", ifName,
+					"-j", "DROP",
+				},
+				protocol: protocol,
+			},
+		}...)
+	}
+	return dropRules
+}
+
+// initExternalBridgeForwardingRules sets up iptables rules for br-* interface svc traffic forwarding
+// -A FORWARD -s 10.96.0.0/16 -j ACCEPT
+// -A FORWARD -d 10.96.0.0/16 -j ACCEPT
+// -A FORWARD -s 169.254.169.1 -j ACCEPT
```

Comment:
can you remind me why we need the .1 here again?

Reply:
ah right cause we will "forward" it to .4 (fake next hop) which OVS will hijack and redirect to OVN directly. Makes sense.